llvm/test/CodeGen/X86/vselect.ll

```diff
@@ -4,6 +4,35 @@
 ; RUN: llc < %s -mtriple=x86_64-unknown-unknown -mattr=+avx | FileCheck %s --check-prefixes=AVX,AVX1
 ; RUN: llc < %s -mtriple=x86_64-unknown-unknown -mattr=+avx2 | FileCheck %s --check-prefixes=AVX,AVX2
 
+; should Fold select (sext m), (add X, C), X --> (add X, (and C, (sext m))))
```

Comment:
Include the reference to #66101 for people looking at this in the future:
`; PR66101 - Fold select (sext m), (add X, C), X --> (add X, (and C, (sext m))))`